src/hevm/src/EVM/Types.hs

```diff
@@ -137,9 +138,7 @@ readN :: Integral a => String -> a
 readN s = fromIntegral (read s :: Integer)
 
 readNull :: Read a => a -> String -> a
-readNull x s = case Text.Read.readMaybe s of
-  Just y  -> y
-  Nothing -> x
+readNull x s = fromMaybe x (Text.Read.readMaybe s)
```

Comment:
can probably be made pointless too?

Reply:
hmmmm, what would that look like exactly?

This fails to compile with `cannot construct the infinite type: a ~ String -> a`:

```
readNull x = fromMaybe x $ Text.Read.readMaybe
```

Reply:
I was thinking:
```
readNull x = fromMaybe x . Text.Read.readMaybe
```
or something like that

Reply:
works! Thanks :heart: